pkg/api/validate/dynamic/serviceprincipal.go

```diff
@@ -0,0 +1,32 @@
+package dynamic
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/form3tech-oss/jwt-go"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclaim"
+)
+
+func (dv *dynamic) ValidateClusterServicePrincipalToken(ctx context.Context, token *adal.ServicePrincipalToken) error {
```

Comment:
surely this should be taking clientid/clientsecret/tenantid so it can get its own graph token?

Reply:
maybe :) it depends really...